src/index.ts

```diff
@@ -170,18 +179,40 @@ async function collectMappingOptions(
   }
 }
 
-async function apply(organizeOptions: OrganizeOptions) {
+export type ProgressCallback = (message: TWorkerMessage) => void
+
+let progressCallback: ProgressCallback | undefined
+
+export async function apply(
+  options: OrganizeOptions,
+  onProgress?: ProgressCallback,
+): Promise<TMapResults[]> {
+  progressCallback = onProgress
+
+  // Initialize workers
   const fileListWorker = initializeFileListWorker()
   initializeMappingWorkers()
-  // Set global mappingWorkerOptions
-  mappingWorkerOptions = (await collectMappingOptions(
-    organizeOptions,
```

Comment:
Did you intentionally rename to `options`? Okay if you did, if not, please revert to avoid noise.